makefile

```diff
@@ -120,26 +119,33 @@ usage:
 	@echo " make fortran - compile and run Fortran examples"
 	@echo " make matlab - compile MATLAB interfaces"
 	@echo " make octave - compile and test octave interfaces"
-	@echo " make python - compile and test python interfaces"	
+	@echo " make python - compile and test python interfaces"
 	@echo " make all - do all the above (around 1 minute; assumes you have MATLAB, etc)"
 	@echo " make spreadtest - compile and run spreader tests only"
 	@echo " make objclean - remove all object files, preserving libs & MEX"
 	@echo " make clean - also remove all lib, MEX, py, and demo executables"
 	@echo "For faster (multicore) making, append the flag -j"
 	@echo ""
-	@echo "Compile options: 'make [task] PREC=SINGLE' for single-precision"
+	@echo "Compile options:"
 	@echo " 'make [task] OMP=OFF' for single-threaded (otherwise OpenMP)"
 	@echo " You must 'make objclean' before changing such options!"
 	@echo ""
 	@echo "Also see docs/install.rst"
 
 # implicit rules for objects (note -o ensures writes to correct dir)
-%.o: %.cpp %.h
```

Comment:
Looks like we're losing dependency on the headers here. Is there some way to reintroduce this in the new framework?

Reply:
Looks can be deceiving...  I don't think that combo pattern would match anything because the .cpp and .h are not in the same directory, so their `%` stems wont match, eg `src/foo.cpp` `include/foo.cpp`, would imply stems `src/foo != include/foo`.  These lines didn't actually work for me as expected until I ditched the %.h.... I think might have been catching other rules/defaults, but enough stuff was defined in that makefile that it worked okay ;) ... I think they must be from a time the code and headers were all in one folder.

Anyway, yes, they should depend on the headers. I can add back a different rule.

Reply:
pushed a patch.  Changing headers will now rebuild all sources. This is a much better rule anyway...